src/json_deser.c

```diff
@@ -401,3 +414,140 @@ corto_int16 json_deserialize(corto_object o, corto_string s) {
     }
     return -1;
 }
+
+/*
+ * This method changes the id!
+ */
+static void json_splitParentAndName(char* id, char** parentName, char** name)
+{
+    char *ptr = strrchr(id, '/');
+    if (ptr) {
+        *name = ptr + 1;
+        *parentName = id;
+        *ptr = '\0';
+    } else {
+        *name = id;
+        *parentName = NULL;
+    }
+}
+
+static corto_object json_declareObjectByParentNameNameTypeName(const char* parentName, const char* name, const char* typeName)
+{
+    corto_object package = NULL;
+    corto_object type = corto_resolve(NULL, (corto_string)typeName);
+    if (!type) {
+        corto_seterr("Cannot find %s", typeName);
+        goto errorTypeNotFound;
+    }
+    if (!corto_instanceof(corto_type_o, type)) {
+        corto_seterr("%s is not a type", typeName);
+        goto errorNotType;
+    }
+    if (parentName) {
```

Comment:
Get rid of this if condition after changing the `json_splitParentAndName` function.